processing/src/main/java/io/druid/query/QueryRunnerFactory.java

```diff
@@ -57,7 +57,7 @@
    * @param queryRunners Individual QueryRunner objects that produce some results
    * @return a QueryRunner that, when asked, will use the ExecutorService to runt he base QueryRunners
    */
-  public QueryRunner<T> mergeRunners(ExecutorService queryExecutor, Iterable<QueryRunner<T>> queryRunners);
+  public QueryRunner<T> mergeRunners(ExecutorService queryExecutor, Iterable<QueryRunner<T>> queryRunners, QueryType query);
```

Comment:
yeah, thought about that but can't find any alternative and this option looked ok to me.